internal/cmd/providers.go

```diff
@@ -28,11 +30,34 @@ func newProvidersCmd(cli *CLI) *cobra.Command {
 
 	cmd.AddCommand(newProvidersListCmd(cli))
 	cmd.AddCommand(newProvidersAddCmd(cli))
+	cmd.AddCommand(newProvidersEditCmd(cli))
 	cmd.AddCommand(newProvidersRemoveCmd(cli))
 
 	return cmd
 }
 
+func newProvidersEditCmd(cli *CLI) *cobra.Command {
+	var secret bool
+
+	cmd := &cobra.Command{
+		Use:   "edit PROVIDER",
+		Short: "Update a provider",
+		Example: `# Set a new secret for a provider
+$ infra providers edit okta --secret`,
+		Args: ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !secret {
+				return errors.New("Please specify a field to update. For options, run 'infra providers edit --help'")
+			}
+
+			return updateProvider(cli, args[0])
+		},
+	}
+
+	cmd.Flags().BoolVar(&secret, "secret", false, "Set a new secret")
```

Comment:
We're missing some values here:

```
--client-secret
--client-id
--url
--kind
```

Reply:
The ticket was only for secret, do we want to support all? I was under the impression that users should not change other fields and should rather create a new provider instead.

Reply:
That makes sense, since `--client-secret` is often rotated. Let's stick to only `--secret`, but we should rename it to `--client-secret`